src/utils/callouts.ts

```diff
@@ -29,58 +50,373 @@ function convertComponentToFilter(
     nullable: true,
   };
 
-  switch (component.type) {
-    case "checkbox":
-      return { ...baseItem, type: "boolean", nullable: false };
-
-    case "number":
-      return { ...baseItem, type: "number" };
-
-    case "select":
-      return {
-        ...baseItem,
-        type: "enum",
-        options: convertValuesToOptions(component.data.values),
-      };
-
-    case "selectboxes":
-    case "radio":
-      return {
-        ...baseItem,
-        type: component.type === "radio" ? "enum" : "array",
-        options: convertValuesToOptions(component.values),
-      };
-
-    case "textarea":
-      return { ...baseItem, type: "blob" };
-
-    default:
-      return { ...baseItem, type: "text" };
+  if (
+    isCalloutInputCheckboxComponent(component)
+  ) {
+    return { ...baseItem, type: "boolean", nullable: false };
   }
+
+  if (isCalloutInputNumberComponent(component)) {
+    return { ...baseItem, type: "number" };
+  }
+
+  if (
+    isCalloutInputSelectComponent(
+      component,
+    )
+  ) {
+    return {
+      ...baseItem,
+      type: "enum",
+      options: convertValuesToOptions(component.data.values),
+    };
+  }
+
+  if (
+    isCalloutInputTextAreaComponent(component) && component.type === "textarea"
+  ) {
+    return { ...baseItem, type: "blob" };
+  }
+
+  return { ...baseItem, type: "text" };
 }
 
-function getNiceAnswer(
+function getSelectableLabelFromValue(
   component: CalloutComponentSchema,
   value: string,
 ): string {
-  switch (component.type) {
-    case "radio":
-    case "selectboxes":
-      return component.values.find((v) => v.value === value)?.label || value;
-    case "select":
-      return (
-        component.data.values.find((v) => v.value === value)?.label || value
-      );
-    default:
-      return value;
+  if (isCalloutInputSelectableComponent(component)) {
+    return component.values.find((v) => v.value === value)?.label || value;
   }
+
+  if (isCalloutInputSelectComponent(component)) {
+    return (
+      component.data.values.find((v) => v.value === value)?.label || value
+    );
+  }
+
+  return value;
+}
+
+export function isCalloutComponentOfType<T extends CalloutComponentBaseSchema>(
```

Comment:
This isn't really type safe as it requires you to specify `T`.

There will be a way of inferring `T` from `type`

```ts
export function isCalloutComponentOfType<T extends CalloutComponentType)>(
  component: CalloutComponentBaseSchema,
  type: T
): component is CalloutComponentMap[T] {
  return "type" in component &&
    component.type === type;
}
```

Where `CalloutComponentMap` is some sort of type mapping

Reply:
Thank you, I will do that